components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -13390,6 +13425,34 @@ public String getAPILevelTier(int id) throws APIManagementException {
         return apiLevelTier;
     }
 
+    public ThrottlingLimit getAPIThrottlingLimit(String uuid) throws APIManagementException {
+
+        ThrottlingLimit apiLevelLimit = null;
+        try (Connection connection = APIMgtDBUtil.getConnection()) {
+            apiLevelLimit = getAPIThrottlingLimit(connection, uuid);
+        } catch (SQLException e) {
+            handleException(String.format("Failed to get API Throttling Limit for the API %s", uuid), e);
+        }
+        return apiLevelLimit;
+    }
+
+    public ThrottlingLimit getAPIThrottlingLimit(Connection connection, String uuid) throws SQLException {
+        String apiLevelTierJson = null;
+        String query = SQLConstants.GET_API_THROTTLE_LIMIT_SQL;
+        try (PreparedStatement selectPreparedStatement
+                     = connection.prepareStatement(query + " WHERE API_UUID = ?")) {
+            selectPreparedStatement.setString(1, uuid);
+            ResultSet resultSet = selectPreparedStatement.executeQuery();
+            while (resultSet.next()) {
+                apiLevelTierJson = resultSet.getString(SQLConstants.ColumnName.API_THROTTLE_LIMIT);
+            }
```

Comment:
resultSet also needs to be added with try with resources
```suggestion
            try (ResultSet resultSet = selectPreparedStatement.executeQuery()) 
                while (resultSet.next()) {
                    apiLevelTierJson = resultSet.getString(SQLConstants.ColumnName.API_THROTTLE_LIMIT);
                }
            }
```